jcl/src/java.base/share/classes/com/ibm/tools/attach/target/CommonDirectory.java

```diff
@@ -126,32 +128,59 @@ static void prepareCommonDirectory() throws IOException {
 	 * Lock the master lock file. Create the lockfile if necessary.
 	 * @throws IOException if the file is already locked.
 	 */
-	/*[PR Jazz 30075] obtainMasterLock(File commonDirectory) is never used. */		
 	public static void obtainMasterLock() throws IOException {
-		getMasterLock().lockFile(true);
+		boolean doLock = false;
+		FileLock masterLockCopy;
+		synchronized (accessorMutex) {
+			++masterLockCount;
+			masterLockCopy = getMasterLock();
+			doLock = (1 == masterLockCount); 
+		}
+		if (doLock) { /* first entry */
+			masterLockCopy.lockFile(true);
+		}
 	}
 
 	/**
-	 * non-blocking lock the master lockfile. Create the lockfile if necessary.
+	 * non-blocking lock the master lockfile. 
+	 * Create the lockfile if necessary.
 	 * @return true if lock obtained
-	 * @throws IOException if file already locked
 	 */
-	static boolean tryObtainMasterLock() throws IOException {
-		return getMasterLock().lockFile(false);
+	static boolean tryObtainMasterLock() {
+		boolean masterLockEntered =true;
+		synchronized (accessorMutex) {
+			++masterLockCount; /* optimistically assume we enter the lock */
+			if (1 == masterLockCount) { /* first time in */
+				try {
+					masterLockEntered = getMasterLock().lockFile(false);
+				} catch (IOException e) {
+					IPC.logMessage("IOException in tryObtainMasterLock, masterLockCount=", masterLockCount); //$NON-NLS-1$
```

Comment:
It would be useful to log the cause of the IOException and the stack trace as well.